internal/loader/settings_client.go

```diff
@@ -11,26 +11,78 @@ import (
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
 	"github.com/Azure/azure-sdk-for-go/sdk/data/azappconfig"
-	"k8s.io/klog/v2"
+	"k8s.io/klog"
 )
 
 //go:generate mockgen -destination=mocks/mock_settings_client.go -package mocks . SettingsClient
 
-type SelectorSettingsClient struct {
-	selectors []acpv1.Selector
+type EtagSettingsClient struct {
+	etags       map[acpv1.Selector][]*azcore.ETag
+	etagChanged bool
```

Comment:
I don't expect to have this field in `EtagSettingsClient`, if the etag is not change, just return nil etgas is fine.